tls/s2n_early_data.h

```diff
@@ -54,3 +54,15 @@ int s2n_psk_configure_early_data(struct s2n_psk *psk, uint32_t max_early_data_si
         uint8_t cipher_suite_first_byte, uint8_t cipher_suite_second_byte);
 int s2n_psk_set_application_protocol(struct s2n_psk *psk, const uint8_t *application_protocol, uint8_t size);
 int s2n_psk_set_context(struct s2n_psk *psk, const uint8_t *context, uint16_t size);
+
+typedef enum {
+    S2N_EARLY_DATA_STATUS_UNKNOWN = 0,
```

Comment:
At what point is the status supposed to be known? How does the caller know when that happens?

Reply:
S2N_EARLY_DATA_STATUS_UNKNOWN should never happen. If you check the implementation, it's only returned for an invalid early data state.

Maybe an error would be a more correct response.